File: backend/lambdas/broadcast-lambda/src/main/java/com/artcorner/erp/dto/response/reports/SalesReportResponse.java
package com.artcorner.erp.dto.response.reports;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Data
@Builder
public class SalesReportResponse {

    private LocalDate fromDate;
    private LocalDate toDate;

    private int totalOrders;
    private int completedOrders;

    private BigDecimal totalRevenue;
    private BigDecimal totalProfit;

    private List<BigDecimal> dailyRevenue;
    private List<BigDecimal> dailyProfit;
    private List<Integer> dailyOrders;
    private List<Integer> dailyCompletedOrders;
}

File: frontend/Admin/src/features/reports/types/SalesReportResponse.ts
export interface SalesReportResponse {
    fromDate: string;
    toDate: string;

    totalOrders: number;
    completedOrders: number;

    totalRevenue: number;
    totalProfit: number;

    dailyRevenue: number[];
    dailyProfit: number[];
    dailyOrders: number[];
    dailyCompletedOrders: number[];
}